mmdet3d/datasets/pipelines/transforms_3d.py

```diff
@@ -1422,3 +1424,261 @@ def _auto_indent(repr_str, indent):
+                           0) & (centers2d[:, 0] <
+                                 self.img_scale[0]) & (centers2d[:, 1] > 0) & (
+                                     centers2d[:, 1] < self.img_scale[1])
+            results['centers2d'] = centers2d[valid_index]
+
+            for key in results.get('bbox_fields', []):
+                if key in ['gt_bboxes']:
+                    results[key] = results[key][valid_index]
+                    if 'gt_labels' in results:
+                        results['gt_labels'] = results['gt_labels'][
+                            valid_index]
+                    if 'gt_masks' in results:
+                        raise NotImplementedError(
+                            'AffineResize only supports bbox.')
+
+            for key in results.get('bbox3d_fields', []):
+                if key in ['gt_bboxes_3d']:
+                    results[key].tensor = results[key].tensor[valid_index]
+                    if 'gt_labels_3d' in results:
+                        results['gt_labels_3d'] = results['gt_labels_3d'][
+                            valid_index]
+
+            results['depths'] = results['depths'][valid_index]
+
+        return results
+
+    def _affine_bboxes(self, results, trans_affine):
+        """Affine transform bboxes to input image.
+
+        Args:
+            results (dict): Result dict from loading pipeline.
+            trans_affine (np.ndarray): Matrix transforming original
+                image to the network input image size.
+        """
+
+        for key in results.get('bbox_fields', []):
+            bboxes = results[key]
+            bboxes[:, :2] = self._affine_transform(bboxes[:, :2], trans_affine)
+            bboxes[:, 2:] = self._affine_transform(bboxes[:, 2:], trans_affine)
+            if self.bbox_clip_border:
+                bboxes[:,
+                       [0, 2]] = bboxes[:,
+                                        [0, 2]].clip(0, self.img_scale[0] - 1)
+                bboxes[:,
+                       [1, 3]] = bboxes[:,
+                                        [1, 3]].clip(0, self.img_scale[1] - 1)
+            results[key] = bboxes
+
+    def _affine_transform(self, point, matrix):
+        """Affine transform bbox points to input iamge.
+
+        Args:
+            points (np.ndarray): Points to be transformed.
+                shape: (N, 2)
+            matrix (np.ndarray): Affine transform matrix.
+                shape: (3, 3)
+
+        Returns:
+            np.ndarray: Transformed points.
+        """
+        point_num = point.shape[0]
+        point_exd = np.concatenate((point, np.ones((point_num, 1))), axis=1)
+        point_exd_trans = point_exd.T
+        new_point = np.matmul(matrix, point_exd_trans).T
+        return new_point[:, :2]
+
+    def _get_transfrom_matrix(self, center_scale, output_size):
+        """Get affine transform matrix.
+
+        Args:
+            center_scale (list[tuple]): Center and scale of
+                current image.
```

Comment:
It is a little strange to merge center and scale into a list. Please consider split them into two variables